python/scripts/build_python_package.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+# Copyright 2023-2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+# This script builds the python package via `uv build`. However, our mixed
+# rust/python setup currently has some issues and the building process fails if
+# rust's cli has a readme. Thus, we delete such readme (only if it is safe to do
+# so), we build the python package, and we then restore the readme.
+
+# From https://stackoverflow.com/a/246128
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+PYTHON_ROOT_DIR=$(realpath "$SCRIPT_DIR/..")
+DIST_OUTPUT_DIR=$PYTHON_ROOT_DIR/dist
+RUST_CLI_README_REL_PATH="rust/cli/README.md"
+RUST_CLI_README_ABS_PATH=$(realpath "$SCRIPT_DIR/../../$RUST_CLI_README_REL_PATH")
+
+# if rust's cli readme has local modifications, abort
+if [[ -n $(git status --porcelain | grep $RUST_CLI_README_REL_PATH) ]]; then
```

Comment:
We could simply `[ -z "$(git status --porcelain)" ]` at the beginning of the script. We shouldn't run this if the repository is dirty right?

Reply:
true, but in practice it's annoying when developing this script itself as the git repo will always be in a dirty state. But it's a good point, will do this in my python rewrite.

Reply:
Yes, in those cases I just comment the check during development (but I also sometimes just create a fake commit with my development changes which is sometimes needed depending on what the script does).